### PR TITLE
Wasm: updates to generated entity classes plus some tidying

### PR DESCRIPTION
**Pull request by mykmartin**
- Entities: support for comparisons and use in STL containers
- Better technique for forcing a compiler error in misused templates
- Internal-ized the encode/decode helpers

### CODE REVIEW

**Comment by mykmartin** [comment]
fyi @cromwellian this is really just an internal update to the C++ code (no wasm API changes)

File: src/tools/schema2packager.ts [hunk 1]
```diff
@@ -163,12 +175,25 @@ ${name} clone_entity(const ${name}& entity) {
 }
 
 template<>
-void decode_entity(${name}* entity, const char* str) {
+bool entities_equal(const ${name}& a, const ${name}& b) {
+  return ${equals.join('\n         ')} 1;
```
Comment (csilvestrini) [inline src/tools/schema2packager.ts]: What's this 1 doing here??

Comment (csilvestrini) [reply src/tools/schema2packager.ts]: OK I figured it out.. That's pretty subtle.

How about `equals.join('\n      &&')` ?

Comment (mykmartin) [reply src/tools/schema2packager.ts]: Updated to put the `&&` s in that line.

**Pull request merged**